hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/model/AbstractHoodieRowData.java

```diff
@@ -59,6 +60,11 @@ public AbstractHoodieRowData(String commitTime,
     this.row = row;
   }
 
+  public void updateMetaField(int pos, String value) {
+    ValidationUtils.checkArgument(pos < metaColumnsNum, "Invalid position for metadata field: " + pos);
+    metaColumns[pos] = value;
```

Comment:
let's try to make the row data immutabe, maybe add a `copy(String[] metaColumns)` for it.